### PR TITLE
Recover unpackaged tests

### PR DESCRIPTION
**Pull request by pramsey**

### CODE REVIEW

File: postgis/postgis.sql.in [hunk 1]
```diff
@@ -2928,11 +2934,24 @@ BEGIN
 							WHERE n.nspname = 'tiger' AND c.relname = 'geocode_settings') )
 			THEN
 				--force install in same schema as postgis only if postgis extension is installed
+				major_version = int4((regexp_match(split_part(version(),' ',2), '^(\d+)'))[1]);
```
Comment (Algunenano) [inline postgis/postgis.sql.in:2937]: I would recommend using `server_version_num` here, e.g. `IF current_setting('server_version_num')::integer >= 130000`

File: postgis/postgis.sql.in [hunk 2]
```diff
@@ -2881,7 +2900,8 @@ BEGIN
 			RAISE DEBUG '%', sql;
 			EXECUTE sql;
 		ELSIF (rec.default_version = rec.installed_version AND rec.installed_version ILIKE '%dev') OR
-			(@extschema@._postgis_pgsql_version() != @extschema@._postgis_scripts_pgsql_version())
+			--(@extschema@._postgis_pgsql_version() != @extschema@._postgis_scripts_pgsql_version())
+			(public._postgis_pgsql_version() != public._postgis_scripts_pgsql_version())
```
Comment (strk) [inline postgis/postgis.sql.in:2972]: Why the hard-coded `public` here instead of keeping @extschema@ ?

File: regress/run_test.pl [hunk 3]
```diff
@@ -718,12 +728,25 @@ sub drop_table
...
+{
+	my $sql = shift;
+	return sql_db($sql, $DB);
+}
+
+sub sql_db_postgres
+{
+	my $sql = shift;
+	return sql_db($sql, "postgres");
 }
```
Comment (strk) [inline regress/run_test.pl:765]: what's this noise with sql_db and sql_postgresl ? Unused ?

File: postgis/postgis.sql.in (hunk 2 above)
Comment (pramsey) [reply postgis/postgis.sql.in:2972]: Good catch. Testing purposes during dev.

File: postgis/postgis.sql.in (hunk 1 above)
Comment (strk) [reply postgis/postgis.sql.in:2937]: See https://dronie.osgeo.org/postgis/postgis/1401/1/3 (fails with 9.6)

**Comment by strk** [comment]
I've updated https://git.osgeo.org/gitea/postgis/postgis/pulls/45 to deal with postgis_extensions_upgrade, with the aim of reducing changes and cleaning up the schema target enforcement too

**Comment by strk** [comment]
This PR can be closed, I pushed my PR with updated postgis_extensions_upgrade as 520af0e8d1bb6e9544d974713bb87e428ff4230b

**Pull request closed without merging**